src/main/java/io/codelex/flightplanner/flightplan/FlightPlanDbService.java

```diff
@@ -0,0 +1,133 @@
+package io.codelex.flightplanner.flightplan;
+
+import io.codelex.flightplanner.flightplan.domain.Airport;
+import io.codelex.flightplanner.flightplan.domain.Flight;
+import io.codelex.flightplanner.flightplan.dto.FlightRequest;
+import io.codelex.flightplanner.flightplan.dto.PageResult;
+import io.codelex.flightplanner.flightplan.dto.SearchFlightsRequest;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+
+@Service
+@ConditionalOnProperty(prefix = "flightplanner", name = "appmode", havingValue = "in-database")
+public class FlightPlanDbService implements FlightPlanService {
+
+    private final FlightRepository flightRepository;
+    private final AirportRepository airportRepository;
+
+    private static final DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+
+    public FlightPlanDbService(FlightRepository flightRepository, AirportRepository airportRepository) {
+        this.flightRepository = flightRepository;
+        this.airportRepository = airportRepository;
+    }
+
+    @Override
+    public Flight addFlight(FlightRequest flightRequest) {
+        Flight newFlight = parseToFlight(flightRequest);
+        if (flightAlreadyAdded(newFlight)) {
+            throw new ResponseStatusException(HttpStatus.CONFLICT, "Can't add same flight twice");
+        } else if (sameAirport(newFlight)) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Same Airports");
+        } else if (!validDatesFlight(newFlight)) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Arrival after departure");
+        } else {
+            this.airportRepository.save(newFlight.getFrom());
+            this.airportRepository.save(newFlight.getTo());
+            return this.flightRepository.save(newFlight);
+        }
+    }
+
+    public Flight parseToFlight(FlightRequest flightRequest) {
+
+        return new Flight(flightRequest.getFrom(),
+                flightRequest.getTo(),
+                flightRequest.getCarrier(),
+                LocalDateTime.parse(flightRequest.getDepartureTime(), dateTimeFormatter),
+                LocalDateTime.parse(flightRequest.getArrivalTime(), dateTimeFormatter));
+
+    }
+
+    public boolean flightAlreadyAdded(Flight flight) {
+
+        return this.flightRepository.findAll().stream()
```

Comment:
Pamanīju vēl šo...
šitā nevajadzētu darīt:
this.flightRepository.findAll()
Ja DB būtu 10 000 ieraksti, tad viņi visu tiktu ielasīti atmiņā un tas var vienā brīdī nokārt sistēmu. Bet ir iespējams uzreiz repo meklēt pēc konkrētiem kritērijiem, vai rakstīt custom meklēšanu, kas notiek DB pusē.
Izlasi šito un arī zemāk ir manual custom queries:
https://www.baeldung.com/the-persistence-layer-with-spring-data-jpa#1-automatic-custom-queries

Tas pats iekš searchAirport un searchFlights